src/scene/axis.js

```diff
@@ -407,19 +417,52 @@ function axisTicksExtend(orient, ticks, oldScale, newScale, size, offset) {
   } else {
     size = {value: sign * size, offset: offset};
   }
+
+  var tickPlacement = config_axis.tickPlacement;
   if (orient === 'top' || orient === 'bottom') {
+    var updatedOldScale = {'scale': oldScale.scale, 'offset': oldScale.offset};
+    var updatedNewScale = {'scale': newScale.scale, 'offset': newScale.offset};
+    if (tickPlacement === 'between' && scaleType === ORDINAL) {
+      var currPadding = 0;
+      var currOuterPadding = 0;
+      for (i=0; i<scales.length; i++) {
+        if (scales[i].name === oldScale.scale) {
+          if ('padding' in scales[i]) {
+           currPadding = scales[i].padding;
+          }
+          if ('outerPadding' in scales[i]) {
+           outerPadding = scales[i].padding;
+          }
+        }
+      }
+      console.log('whattttt');
+      if (!currPadding && typeof config_scale !== 'undefined' && 'padding' in config_scale) {
+        currPadding = config_scale.padding;
+      }
+
+      if (!currOuterPadding && typeof config_scale !== 'undefined' && 'outerPadding' in config_scale) {
+        currOuterPadding = config_scale.outerPadding;
+      }
+
+      var oldPadding = ((oldScale.offset * 2 * currPadding) / ((1 - currPadding) * 2));
+      var newPadding = ((newScale.offset * 2 * currPadding) / ((1 - currPadding) * 2));
+
+      updatedOldScale.offset = ((oldScale.offset - 1) * 2) + oldPadding;
+      updatedNewScale.offset = ((newScale.offset - 1) * 2) + newPadding;
+    }
+
```

Comment:
@emilygu: Could you explain to me what this block of code above is doing? I'm trying to understand why a particular axis needs to check other registered scales?